### PR TITLE
Added Pitch as an alternative sound source

### PR DESCRIPTION
**Pull request by basilefff**
# Objective
My attempt at implementing #7515

## Solution

Added struct `Pitch` and implemented on it `Source` trait.

### Added
- File pitch.rs to bevy_audio crate
- Struct `Pitch` and type aliases for `AudioSourceBundle<Pitch>` and `SpatialAudioSourceBundle<Pitch>`
- New example showing how to use `PitchBundle`

### Changed
- `AudioPlugin` now adds system for `Pitch` audio

### CODE REVIEW

**Comment by mockersf** [comment]
Looks good!

Could you find another name than `Note`? With `Note`, I would expect to give something like `Dsharp` instead of `311.127`.

For audio, could you add a dedicated example in the Audio category instead of changing the breakout one?

**Comment by basilefff** [comment]
> Could you find another name than `Note`? With `Note`, I would expect to give something like `Dsharp` instead of `311.127`.

Sure, I guess the correct name would be pure tone, so maybe `Tone`?

> For the example, could you add a dedicated example in the Audio category instead of changing the breakout one?

Of course, I guess I jumped the gun a little bit because of [this comment](https://github.com/bevyengine/bevy/issues/7515#issuecomment-1418202568)

**Comment by mockersf** [comment]
> Sure, I guess the correct name would be pure tone, so maybe `Tone`?

I think `Pitch` would be the correct name

**Comment by basilefff** [comment]
I changed the name and added example, but I don't know how to fix the failing check. Could you help me with that, please?

**Comment by mockersf** [comment]
The dependencies/check-bans failure is not because of your PR, and it's not required so it can be ignore.

I would love for the example to be a little more interactive. For example react to key press to play a sound, and displaying a text like "press <space> to play a sound". If you're filling more adventurous you could even display the current frequency, and change it up/down on arrow up/down press.

**Comment by basilefff** [comment]
So, something like this with more comments? I think it overcomplicates the example

**Comment by mockersf** [comment]
I think it's great, thanks! if you run `cargo fmt` it should be good to go

**Review by mockersf — APPROVED** [review]

File: examples/README.md [hunk 1]
```diff
@@ -195,6 +195,7 @@ Example | Description
 [Audio](../examples/audio/audio.rs) | Shows how to load and play an audio file
 [Audio Control](../examples/audio/audio_control.rs) | Shows how to load and play an audio file, and control how it's played
 [Decodable](../examples/audio/decodable.rs) | Shows how to create and register a custom audio source by implementing the `Decodable` type.
+[Pitch](../examples/audio/pitch.rs) | Shows how to play a pitch
```
Comment (alice-i-cecile) [inline examples/README.md:198]:
```suggestion
[Pitch](../examples/audio/pitch.rs) | Shows how to directly play a simple pitch
```

File: examples/audio/pitch.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+//! This example illustrates how to play a pitch
```
Comment (alice-i-cecile) [inline examples/audio/pitch.rs:1]:
```suggestion
//! This example illustrates how to play a single-frequency sound (aka a pitch)
```

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
Doc suggestions are just nits to help users who aren't familiar with the terminology. No objections to merging this as is though.

**Pull request merged**